client/src/components/UserMentions.js

```diff
@@ -63,16 +64,40 @@ class UserMentions extends Component {
         super(props);
         this.state = {
             tabValue: 1,
-            mentions: {}
+            page: 0,
+            mentions: {},
+            hasMore: true
         };
     }
+
+    fetchMentions = () => {
+        const { page, mentions } = this.state;
+
+        fetch(MENTIONS_ROUTE + "/" + page, { method: "GET", headers: { "Content-Type": "application/json" } }).then(res => {
+            if (res.status === 200) {
+                res.json().then(data => {
+                    // concatenate the new mentions
+                    let newMentions = mentions;
+                    let numEntries = Object.entries(newMentions).length;
+                    data.forEach((mention, index) => (newMentions[numEntries++] = mention));
+                    this.setState({ mentions: newMentions, page: page + 1 });
+                });
+            } else if (res.status === 204) {
+                // no more mentions to fetch
+                this.setState({ hasMore: false });
+            } else {
+                console.log(res.status, data[RESPONSE_TAG]);
```

Comment:
how will this access the data if it's not within that code block?

Reply:
Good catch. I'll change it